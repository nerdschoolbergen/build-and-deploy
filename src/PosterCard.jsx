import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Typography from '@material-ui/core/Typography';

const styles = {
  card: {
    maxWidth: 500,
  },
  media: {
    height: 750,
  },
};

// This component renders the movie poster and some details about the movie
const PosterCard = ({ classes, posterUrl, movieDetails }) => (
  <div>
    <Card className={classes.card}>
      <CardMedia className={classes.media} image={posterUrl} title={movieDetails.title} />
      <CardContent>
        <Typography variant="h5" component="h2">
          {movieDetails.title}
        </Typography>
        <Typography component="p">{movieDetails.overview}</Typography>
      </CardContent>
      <CardActions>
        {/* TODO */}
      </CardActions>
    </Card>
  </div>
);

PosterCard.propTypes = {
  classes: PropTypes.object.isRequired,
  posterUrl: PropTypes.string.isRequired,
  movieDetails: PropTypes.object.isRequired,
};

export default withStyles(styles)(PosterCard);
