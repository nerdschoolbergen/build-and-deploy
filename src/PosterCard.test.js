import React from 'react';
import renderer from 'react-test-renderer';
import PosterCard from './PosterCard';

describe('PosterCard', () => {
  const defaultProps = {
    posterUrl: '/mock-poster.png',
    movieDetails: {
      title: 'Mock Movie 14',
      description: 'Some movie description',
    },
  };

  it('should render correctly', () => {
    const props = { ...defaultProps };

    // Render the PosterCard component
    const component = renderer.create(
      <PosterCard {...props} />,
    );
    // Expect that its rendered state given these props is equal to the last committed rendered state (snapshot)
    // A snapshot is GUI state represented in code. Think of it like rendering the GUI, taking a picture of how it looks,
    // then the next time we run the test we do the same and compare the pictures. If there are any changes between the two pictures,
    // the test fails, making us manually confirm that the new picture is now the correct state (update the snapshot),
    // or investigate the diff if it was unexpected. Instead of actual pictures we use JSON to compare the gui state. This is snapshot testing.
    expect(component.toJSON()).toMatchSnapshot();
  });
});
