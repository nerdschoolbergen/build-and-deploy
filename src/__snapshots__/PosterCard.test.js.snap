// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PosterCard should render correctly 1`] = `
<div>
  <div
    className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiPaper-rounded-5 MuiCard-root-3 PosterCard-card-1"
  >
    <div
      className="MuiCardMedia-root-31 PosterCard-media-2"
      style={
        Object {
          "backgroundImage": "url(\\"/mock-poster.png\\")",
        }
      }
      title="Mock Movie 14"
    />
    <div
      className="MuiCardContent-root-33"
    >
      <h2
        className="MuiTypography-root-34 MuiTypography-h5-50"
      >
        Mock Movie 14
      </h2>
      <p
        className="MuiTypography-root-34 MuiTypography-body1-43"
      />
    </div>
    <div
      className="MuiCardActions-root-70"
    />
  </div>
</div>
`;
